src/AWS.Deploy.CLI/Commands/TypeHints/VPCConnectorCommand.cs

```diff
@@ -0,0 +1,122 @@
+        }
+
+        private async Task<List<VpcConnector>> GetData()
+        {
+            return await _awsResourceQueryer.DescribeAppRunnerVpcConnectors();
+        }
+
+        public async Task<List<TypeHintResource>?> GetResources(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            var vpcConnectors = await GetData();
+            return vpcConnectors.Select(vpcConnector => new TypeHintResource(vpcConnector.VpcConnectorArn, vpcConnector.VpcConnectorName)).ToList();
+        }
+
+        public async Task<object> Execute(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            _toolInteractiveService.WriteLine();
+            var createNewOptionSetting = optionSetting.ChildOptionSettings.First(x => x.Id.Equals("CreateNew"));
+            var createNew = recommendation.GetOptionSettingValue<string>(createNewOptionSetting) ?? "false";
+            var createNewVPCConnectorAnswer = _consoleUtilities.AskYesNoQuestion("Do you want to create a new VPC Connector?", createNew);
+            var createNewVPCConnector = createNewVPCConnectorAnswer == YesNo.Yes;
+
+            _toolInteractiveService.WriteLine();
+            if (createNewVPCConnector)
+            {
+                _toolInteractiveService.WriteLine("In order to create a new VPC Connector, you need to select 1 or more Subnets as well as 1 or more Security groups.");
+                _toolInteractiveService.WriteLine();
+
+                var availableSubnets = (await _awsResourceQueryer.DescribeSubnets()).OrderBy(x => x.SubnetId).ToList();
+                var userInputConfigurationSubnets = new UserInputConfiguration<Subnet>(
+                    subnet => subnet.SubnetId,
+                    subnet => $"{subnet.SubnetId.PadRight(24)} | {subnet.VpcId.PadRight(21)} | {subnet.AvailabilityZone}",
+                    subnet => false)
+                {
+                    CanBeEmpty = true,
+                    CreateNew = false
+                };
+                var subnetsOptionSetting = optionSetting.ChildOptionSettings.First(x => x.Id.Equals("Subnets"));
+                _toolInteractiveService.WriteLine($"{subnetsOptionSetting.Id}:");
+                _toolInteractiveService.WriteLine(subnetsOptionSetting.Description);
+                var subnets = _consoleUtilities.AskUserForList<Subnet>(userInputConfigurationSubnets, availableSubnets, subnetsOptionSetting, recommendation);
+
+                var availableSecurityGroups = (await _awsResourceQueryer.DescribeSecurityGroups(subnets.ToList())).OrderBy(x => x.VpcId).ToList();
+                var groupNamePadding = 0;
+                availableSecurityGroups.ForEach(x =>
+                {
+                    if (x.GroupName.Length > groupNamePadding)
+                        groupNamePadding = x.GroupName.Length;
+                });
+                var userInputConfigurationSecurityGroups = new UserInputConfiguration<SecurityGroup>(
```

Comment:
If a user is editing the existing subnets and switch to a different VPC that would invalidate the existing security groups. Should we clear out the existing security groups in that scenario to force the user to select a new security group?

Reply:
The control does not force you to select a value for the security groups. So even if we clear them out and the user did not have the intention to edit the security groups, they can still skip them. I think we should leave the behavior as is since a user that is adjusting subnets would keep in mind the security groups.